services/s3/src/it/java/software/amazon/awssdk/services/s3/UploadMultiplePartIntegrationTest.java

```diff
@@ -45,26 +52,89 @@ public static void tearDownFixture() {
 
+        // 3. Complete multipart upload
         CompleteMultipartUploadResponse completeMultipartUploadResponse =
-            s3.completeMultipartUpload(b -> b.bucket(BUCKET).key(KEY).uploadId(multipartUpload.uploadId())
+            s3.completeMultipartUpload(b -> b.bucket(BUCKET)
+                                             .key(key)
+                                             .uploadId(uploadId)
                                              .multipartUpload(CompletedMultipartUpload.builder()
-                                                                                      .parts(CompletedPart.builder().
-                                                                                          eTag(etag).partNumber(1).build())
+                                                                                      .parts(completedParts)
                                                                                       .build()).build());
 
         assertThat(completeMultipartUploadResponse).isNotNull();
+        verifyMultipartUploadResult(key, contentsToUpload);
+    }
+
+    @Test
+    public void uploadMultiplePart_abort() {
+        String key = "uploadMultiplePart_abort";
+
+        // 1. Initiate multipartUpload request
+        String uploadId = initiateMultipartUpload(key);
+        int partCount = 3;
+
+        // 2. Upload each part
+        List<String> contentsToUpload = new ArrayList<>();
+        List<UploadPartResponse> uploadPartResponses = uploadParts(key, uploadId, partCount, contentsToUpload);
+
+        // 3. abort the multipart upload
+        AbortMultipartUploadResponse abortMultipartUploadResponse =
+            s3.abortMultipartUpload(b -> b.bucket(BUCKET).key(key).uploadId(uploadId));
+
+        // Verify no in-progress multipart uploads
+        ListMultipartUploadsResponse listMultipartUploadsResponse = s3.listMultipartUploads(b -> b.bucket(BUCKET));
+
+        List<MultipartUpload> uploads = listMultipartUploadsResponse.uploads();
+
+        assertThat(uploads).isEmpty();
+    }
+
+    private void verifyMultipartUploadResult(String key, List<String> contentsToUpload) throws Exception {
+        ResponseBytes<GetObjectResponse> objectAsBytes = s3.getObject(b -> b.bucket(BUCKET).key(key),
+                                                                      ResponseTransformer.toBytes());
+        String appendedString = contentsToUpload.stream().collect(StringBuilder::new, StringBuilder::append,
+                                                                  StringBuilder::append).toString();
+        assertThat(objectAsBytes.asUtf8String()).isEqualTo(appendedString);
     }
 
+    private List<UploadPartResponse> uploadParts(String key, String uploadId, int partCount, List<String> contentsToUpload) {
+        List<UploadPartResponse> uploadPartResponses = new ArrayList<>();
+
+        for (int i = 0; i < partCount; i++) {
+            String content = RandomStringUtils.randomAscii(1000);
```

Comment:
nit: unless the entire string needs to be random, we can just use the same string for each part

Reply:
+1